ReSharper.FSharp/src/FSharp.Psi/src/Util/FSharpExpressionUtil.cs

```diff
@@ -35,5 +35,106 @@ public static class FSharpExpressionUtil
 
     public static readonly Func<IFSharpExpression, bool> IsSimpleValueExpressionFunc = IsSimpleValueExpression;
     public static readonly Func<IFSharpExpression, bool> IsLiteralExpressionFunc = IsLiteralExpression;
+
+    // TODO: change name
+    public static IFSharpExpression GetOuterMostParentExpression(this IFSharpExpression expression)
+    {
+      IFSharpTreeNode currentExpr = expression;
+      var parent = currentExpr.Parent;
+      while (parent != null)
+      {
+        switch (parent)
+        {
+          case IChameleonExpression:
+            return currentExpr as IFSharpExpression;
+          case ILiteralExpr literalExpr:
+            currentExpr = literalExpr;
+            parent = literalExpr.Parent;
+            break;
+          case IMatchClause matchClause:
+            if (matchClause.Expression == currentExpr)
+            {
+              currentExpr = matchClause;
+              parent = matchClause.Parent;
+            }
+            else
+            {
+              currentExpr = null;
+              parent = null;
+            }
+            break;
+          case ISequentialExpr sequentialExpr:
+            if (sequentialExpr.Expressions.Last() == currentExpr)
+            {
+              currentExpr = sequentialExpr;
+              parent = sequentialExpr.Parent;
+            }
+            else
+            {
+              currentExpr = null;
+              parent = null;
+            }
+            break;
+          case IIfThenElseExpr ifThenElseExpr:
+            if (ifThenElseExpr.ThenExpr == currentExpr || ifThenElseExpr.ElseExpr == currentExpr)
+            {
+              currentExpr = ifThenElseExpr;
+              parent = ifThenElseExpr.Parent;
+            }
```

Comment:
Please note that the overall type of an `if` expression depends on a presence of an `else` branch:
```fsharp
if true then 1 // the overall type is `unit`
if true then 1 else 2 // the overall type is `int`
```